napari/layers/base/base.py

```diff
@@ -226,18 +226,18 @@ class Layer(KeymapProvider, MousemapProvider, ABC):
     * `_basename()`: base/default name of the layer
     """
 
-    _modeclass = Mode
+    _modeclass: Type[BaseMode] = Mode
 
-    _drag_modes = {
+    _drag_modes: Dict[BaseMode, Callable[[Layer, Event], None]] = {
```

Comment:
Then you could use the `Mode` Enum for the typing in the rest of the file; it doesn't need to be so generic because due to the way we're using Enums here, these modes would have to be changed whenever we use a different `Mode` Enum anyways (someone correct me if I'm wrong).
```suggestion
    _drag_modes: Dict[Mode, Callable[[Layer, Event], None]] = {
```

Reply:
I think this will have to be `StringEnum`, because `Labels` overrides `_drag_modes` with it's own entries that aren't of type `napari.layers.base._base_constants.Mode`

Reply:
what if you typed on the `Labels` layer additionally with `napari.layers.labels.**.Mode`?

Reply:
That won't work, because `napari.layers.base._base_constants.Mode` isn't a subclass of `napari.layers.base._base_constants.Mode`.